src/middlewared/middlewared/plugins/iscsi_/global_linux.py

```diff
@@ -96,6 +96,26 @@ def sessions(self, filters, options):
                 sessions.append(session_dict)
         return filter_list(sessions, filters, options)
 
+    @private
+    async def resync_lun_size_for_zvol(self, id):
+        if not await self.middleware.call('service.started', 'iscsitarget'):
+            return
+
+        extents = {i['id']: i for i in await self.middleware.call('iscsi.extent.query', [['enabled', '=', True]])}
+        resync = bool(list(filter(
+            lambda i: i['extent'] in extents and extents[i['extent']]['path'] == f'zvol/{id}',
+            await self.middleware.call('iscsi.targetextent.query')
+        )))
+
+        if resync:
+            cp = await run(['scstadmin', '-resync_dev', f'{id.split("/")[-1]}'], check=False)
```

Comment:
Just curious how does it distinguish `tank/dataset1/vol` and `tank/dataset2/vol`?

Reply:
Great question, I'll need to test.

Reply:
I was doing this incorrectly and have pushed change to resolve this, good catch.